src/genkat.c

```diff
@@ -115,7 +121,7 @@ void internal_kat(const argon2_instance_t *instance, uint32_t pass) {
                     : ARGON2_QWORDS_IN_BLOCK;
 
             for (j = 0; j < how_many_words; ++j)
-                printf("Block %.4u [%3u]: %016llx\n", i, j,
+                printf("Block %.4u [%3u]: %016" PRIx64 "\n", i, j,
```

Comment:
Does this file really need changing? The value is already being cast to `unsigned long long`, so `%llx` should be appropriate. Or did you intend to remove the cast below?

Reply:
I don’t follow - the issue on mingw is that `printf` doesn’t always understand `%llx` and the definition will substitute the appropriate Microsoft `%I64` version if needed.

Reply:
Ah, I see. I didn't expect there are compilers that don't understand the `llx` format...

And I guess since you fallback to `"llx"` on non-mingw, then you can't safely drop the cast to `unsigned long long`... Never mind my comment then.